src/main/java/org/bytecodeparser/instruction/InstructionArguments.java

```diff
@@ -0,0 +1,22 @@
+package org.bytecodeparser.instruction;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class InstructionArguments {
```

Comment:
you have finite number of PrettyPrintTypes, so you can create finite corresponding number of InstructionArguments implementations

Reply:
do you propose to create InstructionArguments classes for each PrettyPrintTypes?

Reply:
yeap

Reply:
but if we do it we can delete PrettyPrintTypes.. ok, lets leave it as it is now